src/query/parser.ts

```diff
@@ -51,15 +52,18 @@ function tryParseAsYaml(raw: string): any {
   }
 }
 
-function parseObject(query: any): Query {
+function parseObject(pendingQuery: any): Query {
+  let query = pendingQuery.parsedObj;
+  
   if (!query.hasOwnProperty("name") || query.name === null) {
     throw new ParsingError("Missing field 'name' in query");
   }
 
   if (!query.hasOwnProperty("filter") || query.filter === null) {
     throw new ParsingError("Missing field 'filter' in query");
   }
-
+  query.filter = query.filter.replace(/{{filename}}/g, pendingQuery.ctx.sourcePath.replace(/.*\//, '').replace(/\.md$/i, '').replace(/\s+/g, '*'));
```

Comment:
This operation might be better done elsewhere. I.e. - `parseObject` validates that the object is the correct shape and returns a `Query`. Then we introduce another function that performs expansions on the `Query`

This enables a clearer delineation of responsibilities and makes it easier for us to test each method in isolation.

Reply:
Also the series of regex replacements on the `sourcePath`, am I right in thinking this grabs the final file path part (i.e. - `foo.md`), drops the extension, and replaces whitespace characters with `*`?

Why do we want to do the last part?